### ISSUE TITLE
Using regular expression can lead to editor freezing

### ISSUE THREAD

**Issue by rapidsnow** [atom/find-and-replace#174]
When trying to find something like [\s*\.]test[\s\.] via regular expressions, the editor freezes while typing. Specifically, it freezes after I close the first set of square brackets. I'm assuming that this is because there are so many different combinations that match the regular expression that it is having trouble enumerating them all.

I'm not sure if this should really be filed under a bug, as it was obviously my fault for forming a regular expression with so many possible matches, but maybe there should be a catch for when the search is taking too long.

**Comment by beingmrkenny** [comment]
I've found this too, on several occasions. Came here to raise a bug, but seeing as there is one already open, I will comment instead. The freezing is merely annoying, unless the editor has unsaved changes in which case, changes are permanently lost, and that can be a huge problem. Most recently I found this with the regex [^=].

The problem seems to be with automatically highlighting text. I'd suggest making automatic highlight an option — I wouldn't mind having to click "Go" to perform the search.
